packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -3,50 +3,197 @@
+
+/// An object that can be used by a stateful widget to obtain the keyboard focus
+/// and to handle keyboard events.
+///
+/// {@template flutter.widgets.focus_manager.focus.advancedWarning}
+/// **If you are not implementing a widget which handles focused input directly
+/// (e.g. implementing your own text field or other widget primitive), or
+/// keeping [FocusNode] references so that you can control the focus from
+/// another part of the application, you typically will not need to manage a
+/// [FocusNode] directly, as it will be handled by the [Focus] and [FocusScope]
+/// widgets or by the widget primitives themselves. Please see those widgets for
+/// typical usage.**
+/// {@endtemplate}
 ///
-/// The focus tree keeps track of which widget is the user's current focus. The
-/// focused widget often listens for keyboard events.
+/// [FocusNode]s are persistent objects which form a "focus tree" that is a
+/// representation of the widgets in the hierarchy which are interested in
+/// focus. They must be managed like other persistent state, which is typically
+/// done by a [StatefulWidget]. A stateful widget that manages a focus node must
+/// call [dispose] from its [State.dispose] method.
 ///
-/// To request focus, find the [FocusScopeNode] for the current [BuildContext]
-/// and call the [FocusScopeNode.requestFocus] method:
+/// ## Hierarchy Management
 ///
-/// ```dart
-/// FocusScope.of(context).requestFocus(focusNode);
-/// ```
+/// Because [FocusNode]s form a sparse representation of the widget tree, they
+/// must be updated whenever the widget tree is rebuilt. This is done by calling
+/// [reparentIfNeeded], usually from a [State.build] method of the widget which
+/// represents the focused region, so that the size and location of the focused
+/// area are correctly represented in its context.
 ///
-/// If your widget requests focus, be sure to call
-/// `FocusScope.of(context).reparentIfNeeded(focusNode);` in your `build`
-/// method to reparent your [FocusNode] if your widget moves from one
-/// location in the tree to another.
+/// [FocusNodes] are organized into "scopes" (see [FocusScopeNode]), which form
+/// sub-trees of nodes which can be traversed as a group. Within a scope, the
+/// most recent nodes to have focus are remembered, and if a node is focused and
+/// then removed, the original node receives focus again.
+///
+/// The focus node hierarchy can be traversed using the [parent], [children],
+/// [ancestors] and [descendants] accessors.
+///
+/// [FocusNode]s are [ChangeNotifier]s, so you can register to receive a
+/// notification when the focus changes. If you are using the [Focus] and
+/// [FocusScope] widgets to manage the nodes, it is often simpler to just
+/// establish an [InheritedWidget] dependency on them by calling [Focus.of] or
+/// [FocusScope.of] instead.
+///
+/// To see the focus tree in the debug console, you can call
```

Comment:
generally prefer the passive voice. We don't know if the person reading the docs is actually the engineer who will be doing things, so "you" may not be accurate.